sql/src/main/java/io/crate/operation/user/UserManager.java

```diff
@@ -38,26 +38,25 @@
      * creates a user
      *
      * @param userName name of the user to create
-     * @param secureHash the password-hash consisting of a SHA-512 hash, salt and num. iterations
```

Comment:
Not sure about partially commenting parameters. When I read JavaDoc, I expect all paramters to be listed there.

Reply:
I removed it because `SHA-512 hash, salt and num.iterations` imho refers to an implementation detail of the SecureHash class. Such comments are hard to maintain and redundant (It should be documented in the SecureHash class). Similarly, something like `@param securehash - the hashed password` doesn't add useful information. 
I prefer it if documentation/code-comments add additional information, not repeat argument names in a more verbose form. The reason being that if there are no "noisy" comments, the comments which are there receive more attention.

This may be a bit controversial - for example the oracle java style guide also mentions to include all params, but at the same time advices to avoid duplicating information. The google style guide doesn't mention if all params should be documented, but also contains the advice to not write comments if they don't add meaningful information.

Reply:
+1 for removing the implementation detail.

How about: `@param hashedPasswd The hashed password as SecureHash to ensure memory cleanup upon garbage collection. Null means no password set` 

Yes, it is more work and yes there is a potential danger of adding redundant information (although this might only be clear sometimes to the original creator of the code). However, to me the times I find an explanation or hint in the Javadoc/comments by far outweighs the potential noise factor.

Btw, I'm sure the other parameter `* @param userName name of the user to create` is also not so useful in your opinion ;) But even there, consider that people might be wondering if the username is case-sensitive or so.

>The google style guide doesn't mention if all params should be documented, but also contains the advice to not write comments if they don't add meaningful information.

This for sure! Commenting is an art and like programming itself you have to learn how to do it properly. I think we have a certain standard already and don't comment too trivially, like typical example:
```
// increase i by one
i++;
```